tests/publiccloud/nvidia.pm

```diff
@@ -18,24 +18,46 @@ sub validate_nvidia {
     validate_script_output("hwinfo --gfxcard", sub { /nVidia.*Tesla T4/mg });
     # Check loaded modules
     assert_script_run("lsmod | grep nvidia", fail_message => "nvidia module not loaded");
-    # Ideally, we should check for nvidia-smi here but it is delivered by nvidia-compute-utils-G06 package, which is not available on PublicCloud
+    assert_script_run("nvidia-smi");
+}
+
+sub install_nvidia_compute {
+    my $version = script_output('rpm -qa --queryformat \'%{VERSION}\n\' nvidia-open-driver-G06-signed-cuda-kmp-default | cut -d "_" -f1 | sort -u | tail -n 1');
+    zypper_call("in --auto-agree-with-licenses nvidia-compute-utils-G06 == ${version}");
+}
+
+sub install_nvidia {
+    zypper_call("in nvidia-open-driver-G06-signed-kmp-default kernel-firmware-nvidia-gspx-G06 -nvidia-open-driver-G06-signed-cuda-kmp-default -kernel-firmware-nvidia-gspx-G06-cuda", quiet => 1);
+    install_nvidia_compute;
+}
+
+sub install_nvidia_cuda {
+    zypper_call("in nvidia-open-driver-G06-signed-cuda-kmp-default kernel-firmware-nvidia-gspx-G06-cuda -nvidia-open-driver-G06-signed-kmp-default -kernel-firmware-nvidia-gspx-G06", quiet => 1);
+    install_nvidia_compute;
 }
 
 sub run {
     my ($self, $args) = @_;
     script_run("cat /etc/os-release");
+
     if (get_var('NVIDIA_REPO')) {
         zypper_call('--gpg-auto-import-keys addrepo -p 90 ' . get_required_var('NVIDIA_REPO') . ' nvidia_repo');
         zypper_call '--gpg-auto-import-keys ref';
     }
+    if (get_var('NVIDIA_CUDA_REPO')) {
```

Comment:
I know it is done that way above but nevertheless it does not make sense that way . 
```
if(get_var(VAR1)) {
    get_required_var(VAR1)
}
```
so either you want you test to fail when variable is not defined so than dropping check if variable exists or you supporting the case when variable is not defined but than if you enter inside `if` no point to use `get_required_var` because you already KNOW that it exists

Reply:
You can also do   `if (my $nvidia_cuda_repo = get_required_var('NVIDIA_CUDA_REPO'))`

Reply:
TD;LR `NVIDIA_CUDA_REPO` and `NVIDIA_REPO` is requirement for this test and suppose to die if one of them is not defined

Reply:
I like trolling @asmorodskyi so here it comes :-)
TD;LR stands for Too Long Didn't Read.
You got it opposite.  Initially you wrote the long comment. Then added TD;LR which is very short :D